### ISSUE TITLE
Segfault when creating MultiPoint from sequence including empty point 

### ISSUE THREAD

**Issue by jorisvandenbossche** [pygeos/pygeos#134]
Triggered by a similar Shapely issue (https://github.com/Toblerity/Shapely/pull/898, https://github.com/Toblerity/Shapely/issues/836), I wondered how this would work in pygeos.

It's not that easy to create an empty point, so using a workaround with intersection:

```
In [1]: import pygeos 

In [8]: p = pygeos.points(1,1) 

In [9]: box = pygeos.box(2,2,4,4)  

In [10]: p_empty = pygeos.intersection(p, box)     

In [11]: p_empty  
Out[11]: <pygeos.Geometry POINT EMPTY>
```

Then trying to create a MultiPoint from a list that includes this empty point segfaults:

```
In [12]: pygeos.multipoints([p, p_empty])  
Out[12]: Segmentation fault (core dumped)
```

Creating one from just an empty point works, resulting in an empty MultiPoint:

```
In [5]: pygeos.multipoints([p_empty]) 
Out[5]: <pygeos.Geometry MULTIPOINT EMPTY>
```

**Comment by caspervdw** [comment]
Is it the creation of the multipoint, or the `repr` (wkt conversion) that segfaults?

**Comment by jorisvandenbossche** [comment]
Good point :) Indeed, it's the WKT that segfaults, so might be a GEOS issue

**Comment by jorisvandenbossche** [comment]
Converting to WKB raises an informative error message:

```
In [2]: mp_empty  = pygeos.multipoints([p, p_empty])    

In [3]: pygeos.to_wkb(mp_empty)      
...
GEOSException: IllegalArgumentException: Empty Points cannot be represented in WKB
```

**Comment by caspervdw** [comment]
As this is fixed in GEOS >= 3.7 we just need a check here to prevent a segfault for GEOS <3.7
